.github/workflows/ci.yaml

```diff
@@ -28,3 +28,13 @@ jobs:
 
       - name: Execute tests
         run: npm test
+
```

Comment:
Is it necessary to build the image without pushing it into any registry? This step is already done in `cd.yaml`

Reply:
the reason why we put the build step also in the CI pipeline is that we want to make sure that when we open a PR, every change made are built into a container image

Reply:
The container in this step will disappear since it's not stored, so it doesn't matter which changes you made, they will not saved

Reply:
I think this step is useful to spot any problem with `docker build` command before merging a branch to the main branch